### PR TITLE
Dissipation constant fix (#81)

### LINKED ISSUE #81: Dissipation constant should be scaled with mass instead of lengths
**Describe the bug**
In Elastica `dissipation constant (nu)` is defined as per unit length  and its unit is `kg/(m.s)`. However, this assumes that rods are uniform. However, in the case of tapered rods `dissipation constant` should also be scaled with the area. Otherwise for two elements that have same velocity but different mass we apply same dissipation force.

**How to fix**
1. Define the dissipation constant as `1/s` .
2. In `allocate` function,  scale the user defined `nu` with `element mass` and store it.
3. Remove multiplication by `lengths` while computing the dissipation forces and torques.
https://github.com/GazzolaLab/PyElastica/blob/fee87b9da22e310ff925c16fdc839bf8405c51a4/elastica/rod/cosserat_rod.py#L748
https://github.com/GazzolaLab/PyElastica/blob/fee87b9da22e310ff925c16fdc839bf8405c51a4/elastica/rod/cosserat_rod.py#L843

### PR DESCRIPTION
**Pull request by bhosale2**
Fixes #81, as per our discussion @armantekinalp this PR is WIP and we will merge once the dissipation constants are rescaled for examples, and all examples with `nu` have been confirmed to run fine. Other than that, please go through PR and let me know if any changes are needed.

### CODE REVIEW

**Comment by armantekinalp** [comment]
Can we fast-forward `patch_dissipation` branch to `update-0.2.3` ? I think  `patch_dissipation` is created from the `master`  and this PR shows new commits which are already committed in `update-0.2.3`

**Comment by armantekinalp** [comment]
I ran the cases on cluster, and following ones require tuning:
- Helical buckling case exploded for larger than 100 elements.
- Muscular flagella shows a difference with elastica cpp implementation
- Continuum snake velocity profile is not periodic.
- Muscular snake case exploded.
- Rod self contact cases (solenoid and plectoneme) exploded.

**Comment by bhosale2** [comment]
@armantekinalp as discussed, we now decide to premultiply both the velocity and damping terms by mass (to include tapering effects), as included in 89b7bc8.

(List below needs to be checked with nu scaled as `nu = old_nu / density / base_cs_area`)

- [ ] axial stretching
- [ ] binder stuff?
- [ ] continuum flagella
- [ ] continuum snake
- [ ] Experimental cases - parallel connection
- [ ] friction validation cases
- [ ] helical buckling
- [ ] joint cases
- [ ] muscular flagella
- [ ] muscular snake
- [ ] rod-rod contact cases
- [ ] rod self contact cases
- [ ] timoshenko

File: elastica/memory_block/memory_block_rod.py [hunk 1]
```diff
@@ -206,8 +223,9 @@ def allocate_block_variables_in_elements(self, systems: Sequence):
             "rest_lengths": 4,
             "dilatation": 5,
             "dilatation_rate": 6,
-            "dissipation_constant_for_forces": 7,
-            "dissipation_constant_for_torques": 8,
+            # "dissipation_constant_for_forces": 7,
+            # "dissipation_constant_for_torques": 8,
```
Comment (armantekinalp) [inline elastica/memory_block/memory_block_rod.py:227]: maybe just remove these lines

File: elastica/memory_block/memory_block_rod.py [hunk 2]
```diff
@@ -129,17 +129,34 @@ def allocate_block_variables_in_nodes(self, systems: Sequence):
...
+                    self.__dict__[k][..., start_idx:end_idx]
+                )
+
+        # self.mass = self.scalar_dofs_in_rod_nodes[0]
+        # for system_idx, system in enumerate(systems):
+        #     start_idx = self.start_idx_in_rod_nodes[system_idx]
+        #     end_idx = self.end_idx_in_rod_nodes[system_idx]
+        #     self.mass[start_idx:end_idx] = system.mass.copy()
+        #     # create a view back to the rod after copying variable into the block structure
+        #     system.mass = np.ndarray.view(self.mass[start_idx:end_idx])
```
Comment (armantekinalp) [inline elastica/memory_block/memory_block_rod.py:159]: We can remove this as well

**Review by armantekinalp — COMMENTED** [review]
LGTM, just two minor changes

File: elastica/memory_block/memory_block_rod.py (hunk 1 above)
Comment (bhosale2) [reply elastica/memory_block/memory_block_rod.py]: 632b259

File: elastica/memory_block/memory_block_rod.py (hunk 2 above)
Comment (bhosale2) [reply elastica/memory_block/memory_block_rod.py]: 632b259

**Review by armantekinalp — APPROVED** [review]
LGTM

**Pull request merged**